src/type.zig

```diff
@@ -1605,7 +1605,22 @@ pub const Type = struct {
             .struct_type => |struct_type| {
                 if (struct_type.layout == .Packed) {
                     if (opt_sema) |sema| try sema.resolveTypeLayout(ty);
-                    return try Type.fromInterned(struct_type.backingIntType(ip).*).bitSizeAdvanced(mod, opt_sema);
+
+                    const backing_int_type = struct_type.backingIntType(ip).*;
+
+                    if (backing_int_type != .none) {
+                        return try Type.fromInterned(backing_int_type).bitSizeAdvanced(mod, opt_sema);
+                    } else {
+                        assert(struct_type.haveFieldTypes(ip));
+
+                        var size: u64 = 0;
+                        for (0..struct_type.field_types.len) |field_index| {
+                            const field_ty = struct_type.field_types.get(ip)[field_index];
+                            size += try bitSizeAdvanced(Type.fromInterned(field_ty), mod, opt_sema);
+                        }
+
+                        return size;
```

Comment:
In the docs for `backingIntType` I see this:

```zig
        /// The backing integer type of the packed struct. Whether zig chooses
        /// this type or the user specifies it, it is stored here. This will be
        /// set to `none` until the layout is resolved.
        /// Asserts the struct is packed.
        pub fn backingIntType(s: @This(), ip: *const InternPool) *Index {
```

So it looks like instead of duplicating the resolution logic here, it should use the already existing mechanism to ensure the backing integer type.

It looks like that already was being intended to happen by the line above:

```zig
if (opt_sema) |sema| try sema.resolveTypeLayout(ty);
```

So your task is to figure out why calling `resolveTypeLayout` did not result in the backing int type being resolved.